public/app/features/query/components/QueryEditorRows.tsx

```diff
@@ -67,21 +85,29 @@ export class QueryEditorRows extends PureComponent<Props> {
           {(provided) => {
             return (
               <div ref={provided.innerRef} {...provided.droppableProps}>
-                {queries.map((query, index) => (
-                  <QueryEditorRow
-                    dsSettings={dsSettings}
-                    id={query.refId}
-                    index={index}
-                    key={query.refId}
-                    data={data}
-                    query={query}
-                    onChange={(query) => this.onChangeQuery(query, index)}
-                    onRemoveQuery={this.onRemoveQuery}
-                    onAddQuery={this.props.onAddQuery}
-                    onRunQuery={this.props.onRunQueries}
-                    queries={queries}
-                  />
-                ))}
+                {queries.map((query, index) => {
+                  const dataSourceSettings = getDataSourceSettings(query, dsSettings);
+                  const onChangeDataSourceSettings = dsSettings.meta.mixed
```

Comment:
@torkelo currently I check if the callback is set to know if I should render the data source picker. I'm leaning towards adding another boolean flag to the component to control this to prevent writing code like the one I'm referencing here. What is your thoughts on this?